fdbserver/LogSystem.h

```diff
@@ -995,6 +995,22 @@ struct LogPushData : NonCopyable {
 		writtenLocations.clear();
 	}
 
+	// copy next_message_tags into given set
+	void saveTags(std::set<Tag>& tagSet) {
+		tagSet.insert(next_message_tags.begin(), next_message_tags.end());
+		return;
+	}
+
+	// store tlogs as represented by index
+	// also store in tag set all replicas
+	void saveLocations(std::set<uint16_t>& locSet, std::set<Tag>& tagSet) {
+		locSet.insert(msg_locations.begin(), msg_locations.end());
+		for (auto loc: msg_locations) {
+			tagSet.insert(Tag(0, loc));  // TODO POST DEMO support DC other than primary
```

Comment:
the version vector must have updated versions for each SS in a key's team. I am deriving the SS's tag here from the tlog index to get the tag.id, and for the demo, we'll assume locality is 0.

Reply:
after some discussion, this code will likely be changed.